src/_P121_HMC5883L.ino

```diff
@@ -0,0 +1,158 @@
+    Device[deviceCount].Ports = 0;
+    Device[deviceCount].PullUpOption = false;
+    Device[deviceCount].InverseLogicOption = false;
+    Device[deviceCount].FormulaOption = true;
+    Device[deviceCount].ValueCount = 4;
+    Device[deviceCount].SendDataOption = true;
+    Device[deviceCount].TimerOption = true;
+    Device[deviceCount].GlobalSyncOption = true;
+    success = true;
+    break;
+  }
+
+  case PLUGIN_GET_DEVICENAME:
+  {
+    string = F(PLUGIN_NAME_121);
+    success = true;
+    break;
+  }
+
+  case PLUGIN_GET_DEVICEVALUENAMES:
+  {
+    strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_121));
+    strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[1], PSTR(PLUGIN_VALUENAME2_121));
+    strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[2], PSTR(PLUGIN_VALUENAME3_121));
+    strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[3], PSTR(PLUGIN_VALUENAME4_121));
+    success = true;
+    break;
+  }
+  case PLUGIN_WEBFORM_SHOW_I2C_PARAMS:
+  {
+    success = true;
+    break;
+  }
+  case PLUGIN_I2C_HAS_ADDRESS:
+  {
+    success = (event->Par1 == 0x1E);
+    break;
+  }
+  case PLUGIN_WEBFORM_LOAD:
+  {
+    addFormFloatNumberBox(F("DeclinationAngle"), F("plugin_121_HMC5883L_decl"), PCONFIG(0), -2.0f,2.0f ,5, 0.00001f);
+    addUnit(F("degree"));
+    success = true;
+    break;
+  }
+
+  case PLUGIN_WEBFORM_SAVE:
+  {
+    PCONFIG(0) = getFormItemFloat(F("plugin_121_HMC5883L_decl"));
+    success = true;
+    break;
+  }
+
+  case PLUGIN_INIT:
+  {
+    initPluginTaskData(event->TaskIndex, new (std::nothrow) P121_data_struct());
+    P121_data_struct *P121_data =
+        static_cast<P121_data_struct *>(getPluginTaskData(event->TaskIndex));
+
+    if (nullptr != P121_data)
+    {
+      P121_data->initialized = false; // Force re-init just in case the address changed.
+      P121_data->begin();
+      success = P121_data->initialized;
+    }
+    break;
+  }
+  case PLUGIN_READ:
+  {
+    P121_data_struct *P121_data =
+        static_cast<P121_data_struct *>(getPluginTaskData(event->TaskIndex));
+
+    if (nullptr != P121_data)
+    {
+      P121_data->begin();
+
+      if (!P121_data->initialized)
+      {
+        addLog(LOG_LEVEL_ERROR, "Could not initialize HCM5883L");
+        break;
+      }
+      sensors_event_t s_event;
+      P121_data->mag.getEvent(&s_event);
+      UserVar[event->BaseVarIndex + 0] = s_event.magnetic.x;
+      UserVar[event->BaseVarIndex + 1] = s_event.magnetic.y;
+      UserVar[event->BaseVarIndex + 2] = s_event.magnetic.z;
+
+      float heading = atan2(s_event.magnetic.y, s_event.magnetic.x);
+
+      const float decl = PCONFIG(0);
+      addLog(LOG_LEVEL_DEBUG, "Declination Degree:" + String(decl));
+      if (decl != 0)
+      {
+        heading += decl;
+      }
+
+      if (heading < 0)
+        heading += 2*PI;
+
+      if (heading > 2*PI)
+        heading -= 2*PI;
+
+      float degreeHeading = heading * 180/M_PI;
+            UserVar[event->BaseVarIndex + 3] = degreeHeading;
```

Comment:
Can you run uncrustify on it, using the uncrustify config already present in the ESPEasy branch?
You should only need to install the Uncrustify extension in VS code to make it work.
Then right-click on the source code and select to format the document.